web/utils.js

```diff
@@ -439,3 +439,173 @@ function checkEnter(event) {
         $("#sbox").submit();
     }
 }
+
+// Intelligence Window code starts from here
+document.onmousemove = function(event) {
+    if (typeof event !== 'undefined') {
+        document.intelliWindowMouseX = event.clientX;
+        document.intelliWindowMouseY = event.clientY;
+    }
+};
+
+$(document).keypress(function(e) {
+    if (document.activeElement.id === 'search' ||
+        typeof document.intelliWindow === 'undefined') {
+        return true;
+    }
+
+    if (e.which === 49) { // '1' pressed
+        if (document.intelliWindow.className === "intelli_window_style") {
+            hideIntelliWindow();
+        } else if (document.intelliWindow.className === "intelli_window_style_hide") {
+            showIntelliWindow();
+        }
+    }
+    if (e.which === 50) { // '2' pressed
+        var symbol = document.intelliWindow.symbol;
+        var highlighted_symbols_with_same_name = $("a").filter(function(index) {
+            return $(this).text() === symbol &&
+                $(this).css("background-color") === "gold";
+        })
+        if (highlighted_symbols_with_same_name.length === 0) {
+            highlightSymbol(symbol);
+        } else {
+            unhighlightSymbol(symbol);
+        }
+    }
+    return true;
+});
+
+function onMouseOverSymbol(symbol, symbolType) {
+    updateIntelliWindow(symbol, symbolType);
```

Comment:
Hi Martin,

I'm glad you like it.  I, too, have been thinking how to make users aware
of this feature :-)

I'm afraid that highlighting symbol under cursor automatically would be
distracting.  For example, if I want to highlight a symbol, now I need to
be very careful not to point to any other symbols on my way of moving my
cursor towards the symbol I care about; I also need to be careful when
moving my cursor away.  This can be hard if my target is buried in a
cluster of code.

To advertise this, we can add this to the feature list on the OpenGrok
website, to the release notes, and to the help page.

On Sat, Mar 7, 2015 at 3:37 AM, Martin Dobroucký <notifications@github.com>
wrote:

> In web/utils.js
> <https://github.com/OpenGrok/OpenGrok/pull/884#discussion_r25995360>:
>
> > +        var symbol = document.intelliWindow.symbol;
> > +        var highlighted_symbols_with_same_name = $("a").filter(function(index) {
> > +            return $(this).text() === symbol &&
> > +                $(this).css("background-color") === "gold";
> > +        })
> > +        if (highlighted_symbols_with_same_name.length === 0) {
> > +            highlightSymbol(symbol);
> > +        } else {
> > +            unhighlightSymbol(symbol);
> > +        }
> > +    }
> > +    return true;
> > +});
> > +
> > +function onMouseOverSymbol(symbol, symbolType) {
> > +    updateIntelliWindow(symbol, symbolType);
>
> Symbol highlighting feature is fantastic!
> But I'm afraid nobody will ever discover this feature is hidden under 1,2
> key press.
>
> What about automatic highlighting symbol under cursor:
> function onMouseOverSymbol(symbol, symbolType) {
> updateIntelliWindow(symbol, symbolType);
>
> //highlight symbol under cursor
> unhighlightAll();
> highlightSymbol(symbol);
>
> }
>
> I tested this change and it works really nice.
> It's not much annoying for user that don't care about this and it's very
> helpful for anyone trying to read the code in browser.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenGrok/OpenGrok/pull/884/files#r25995360>.
>